src/renderers/shared/fiber/__tests__/ReactIncrementalUpdates-test.js

```diff
@@ -115,141 +124,112 @@ describe('ReactIncrementalUpdates', () => {
-        ops.push('componentDidUpdate');
-      }
       render() {
-        ops.push('render');
         instance = this;
-        return <div />;
+        return <span prop={Object.keys(this.state).sort().join('')} />;
       }
     }
 
     ReactNoop.render(<Foo />);
     ReactNoop.flush();
 
-    ops = [];
-
-    let progressedUpdates = [];
     function createUpdate(letter) {
       return () => {
-        progressedUpdates.push(letter);
+        ReactNoop.yield(letter);
         return {
           [letter]: letter,
         };
       };
     }
 
+    // Schedule some async updates
     instance.setState(createUpdate('a'));
     instance.setState(createUpdate('b'));
     instance.setState(createUpdate('c'));
 
-    // Do just enough work to begin the update but not enough to flush it
-    ReactNoop.flushDeferredPri(15);
-    // expect(ReactNoop.getChildren()).toEqual([span('')]);
-    expect(ops).toEqual(['render']);
-    expect(progressedUpdates).toEqual(['a', 'b', 'c']);
-    expect(instance.state).toEqual({a: 'a', b: 'b', c: 'c'});
-
-    ops = [];
-    progressedUpdates = [];
-
-    instance.setState(createUpdate('f'));
-    ReactNoop.performAnimationWork(() => {
-      instance.setState(createUpdate('d'));
-      instance.setState(createUpdate('e'));
+    // Begin the updates but don't flush them yet
+    ReactNoop.flushThrough(['a', 'b', 'c']);
+    expect(ReactNoop.getChildren()).toEqual([span('')]);
+
+    // Schedule some more updates at different priorities
+    ReactNoop.batchedUpdates(() => {
+      instance.setState(createUpdate('f'));
+      ReactNoop.syncUpdates(() => {
+        instance.setState(createUpdate('d'));
+        instance.setState(createUpdate('e'));
+      });
+      instance.setState(createUpdate('g'));
     });
-    instance.setState(createUpdate('g'));
 
-    ReactNoop.flushAnimationPri();
-    expect(ops).toEqual([
-      // Flushes animation work (d and e)
-      'render',
-      'componentDidUpdate',
-    ]);
-    ops = [];
-    ReactNoop.flush();
-    expect(ops).toEqual([
-      // Flushes deferred work (f and g)
-      'render',
-      'componentDidUpdate',
-    ]);
-    expect(progressedUpdates).toEqual(['d', 'e', 'a', 'b', 'c', 'f', 'g']);
-    expect(instance.state).toEqual({
-      a: 'a',
-      b: 'b',
-      c: 'c',
-      d: 'd',
-      e: 'e',
-      f: 'f',
-      g: 'g',
-    });
+    // The sync updates should have flushed, but not the async ones
+    expect(ReactNoop.getChildren()).toEqual([span('de')]);
+
+    // Now flush the remaining work. Updates a, b, and c should be processed
+    // again, since they were interrupted last time.
+    expect(ReactNoop.flush()).toEqual(['a', 'b', 'c', 'f', 'g']);
+    expect(ReactNoop.getChildren()).toEqual([span('abcdefg')]);
   });
 
   it('can abort an update, schedule a replaceState, and resume', () => {
     let instance;
-    let ops = [];
     class Foo extends React.Component {
       state = {};
-      componentDidUpdate() {
```

Comment:
Why delete this from the assertion? It has nothing to do with the `ReactNoop.yield` stuff so you've lost a part of the test, no?